packages/rest/src/rest.server.ts

```diff
@@ -703,9 +714,66 @@ export class RestServer extends Context implements Server, HttpServerLike {
   }
 }
 
+/**
+ * The form of OpenAPI specs to be served
+ *
+ * @interface OpenApiSpecForm
+ */
+export interface OpenApiSpecForm {
+  version?: string;
+  format?: string;
+}
+
+const OPENAPI_SPEC_MAPPING: {[key: string]: OpenApiSpecForm} = {
+  '/openapi.json': {version: '3.0.0', format: 'json'},
+  '/openapi.yaml': {version: '3.0.0', format: 'yaml'},
+};
+
+/**
+ * Options to customize how OpenAPI specs are served
+ */
+export interface OpenApiSpecOptions {
+  /**
+   * Mapping of urls to spec forms, by default:
+   * ```
+   * {
+   *   '/openapi.json': {version: '3.0.0', format: 'json'},
+   *   '/openapi.yaml': {version: '3.0.0', format: 'yaml'},
+   * }
+   * ```
+   */
+  endpointMapping?: {[key: string]: OpenApiSpecForm};
+
+  /**
+   * A flag to force `servers` to be set from the http request for the OpenAPI
+   * spec
+   */
+  setServersFromRequest?: boolean;
+
+  /**
+   * Configure servers for OpenAPI spec
+   */
+  servers?: ServerObject[];
+}
+
+export interface ApiExplorerOptions {
+  /**
+   * The url for hosted API explorer UI
+   * default to https://loopback.io/api-explorer
+   */
+  url?: string;
+  /**
+   * URL for the API explorer served over plain http to deal with mixed content
+   * security imposed by browsers as the spec is exposed over `http` by default.
+   * https://github.com/strongloop/loopback-next/issues/1603
+   */
+  urlForHttp?: string;
```

Comment:
I would expect LoopBack to be smart enough to figure out that `urlForHttp` is the same as `url`, but using `http://` instead of `https://`.

What are the use cases you have in mind where `urlForHttp` would significantly differ from `url`?

Reply:
`urlForHttp` is optional and we can infer it from `url` in most cases. I added it there since we logically need two values and have to store both of them anyway.

Reply:
How about `httpUrl`?